### ISSUE TITLE
Fix GCC Wreorder warnings

### ISSUE THREAD

**Issue by ice0** [synfig/synfig#1744]
For example:
```
synfig/synfig-studio/src/gui/workarea.h:209:7: warning: ‘studio::WorkArea::onion_skin’ will be initialized after [-Wreorder]
```

**Comment by Keyikedalube** [comment]
- Use CMake to build
Output is clearer and more readable
- Do a clean build
Otherwise use `cmake --build . --target clean` to clean your previous build files
- Apply your OOP knowledge
Good luck :)

**Comment by DhairyaBahl** [comment]
@ice0 would like to work on this issue. Kindly assign it to me. 

Thanks :)

**Comment by ice0** [comment]
@DhairyaBahl Assigned to you! Thank you!

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube This time i did it by synfig::warning("‘studio::WorkArea::onion_skin’ will be initialized after [-Wreorder]");

Is it good ? And I totally understand the topic in detail which you've written in my PR

Thanks :)

**Comment by rodolforg** [comment]
The idea is to **fix** the compiler warning, and not to add a runtime warning ;)
Here's an example: https://stackoverflow.com/questions/1564937/gcc-warning-will-be-initialized-after?rq=1

**Comment by DhairyaBahl** [comment]
@rodolforg Do i have to add -Wno-reorder ? Thanks :)

**Comment by DhairyaBahl** [comment]
I am able to understand that onion_skin has to be declared after some X variable which is dependent on onion_skin but I am unable to solve this. I know its easy but i am getting confused in large codebase. :(

Thanks

**Comment by rodolforg** [comment]
> @rodolforg Do i have to add -Wno-reorder ? Thanks :)

This way you would hide the warning, not fix it ;)



> I am able to understand that onion_skin has to be declared after some X variable which is dependent on onion_skin but I am unable to solve this. I know its easy but i am getting confused in large codebase. :(

Yes, that's exactly it. But you don't have to look the entire codebase: the problem is limited to the class only. The class properties/members should be initialized in constructor initialization list in  the same order they are declared/listed in class definition.

**Comment by DhairyaBahl** [comment]
@rodolforg Okay...Thanks... working on it :)

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube @rodolforg I am unable to find rest of the wreorder warnings. My IDEs are not able to run the the synfig code and I've built the code twice in last 2 days. I am stuck what to do ?? I've tried VS code, QT, netbeans for debugging but none of them are working. IDK what to do??

Kindly help !!

Thanks :)

**Comment by Keyikedalube** [comment]
Have you tried "Clean project" (or similar menu/option name) from your IDE and do a fresh build?

**Comment by Keyikedalube** [comment]
@DhairyaBahl Like this 👇📽👇
[Screencast from 13-10-20 06:09:17 AM IST.zip](https://github.com/synfig/synfig/files/5368286/Screencast.from.13-10-20.06.09.17.AM.IST.zip)

That way you'll again notice all the warning messages like the first time you build synfig.

**Hint:** After build finishes just click inside the "Compile Output" panel and hit `Ctrl+F`. Use that to track down all occurences of the word "Wreorder"

**Comment by ice0** [comment]
> I've tried VS code, QT, netbeans for debugging but none of them are working.

That's interesting. Which OS you are using?

**Comment by DhairyaBahl** [comment]
@ice0 windows 

@Keyikedalube I've built it in mingw64,do I have to build it in ide also? 

Thanks :)

**Comment by Keyikedalube** [comment]
> do I have to build it in ide also?

No. You can either use command line or IDE. Both works.

Are you using Autotools or CMake?

**Comment by DhairyaBahl** [comment]
@Keyikedalube I built it using cmake and I can run the built synfigstudio.exe file

**Comment by DhairyaBahl** [comment]
@Keyikedalube but IDK where to get errors and warnings

**Comment by Keyikedalube** [comment]
build-cmake.sh script or your own generator?

**Comment by Keyikedalube** [comment]
@DhairyaBahl  Do you know your build output directory then?

**Comment by DhairyaBahl** [comment]
@Keyikedalube yes I know.. it's somewhere in output bin... I can find it so yes I know

**Comment by Keyikedalube** [comment]
If possible just send me a screenshot of your `synfig/` directory please

**Comment by Keyikedalube** [comment]
@DhairyaBahl try to fit all the folder contents in one shot by reducing folder icon size like this

![Screenshot from 2020-10-13 10-03-34](https://user-images.githubusercontent.com/24214509/95815907-642f8700-0d3b-11eb-8ec3-eae224087560.png)

**Comment by DhairyaBahl** [comment]
@Keyikedalube what is this ?? What do u mean by reducing..??  This is not the normal forked directory

**Comment by DhairyaBahl** [comment]
Should I delete all the files except these in my local directory? Or in build directory.. I am not getting ur point @Keyikedalube

**Comment by Keyikedalube** [comment]
@DhairyaBahl I mean your project root directory you cloned.

Anyway, in my screenshot do you see `build-synfig-Synfig-Debug` folder? Now, this folder is my build directory and yours probably will be under a different name. Delete your build folder and build synfig again. 

> Or in build directory

Delete that build directory.

If you send me your folder screenshot, it'd make our complicated conversation much easier...

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okay just gimme a sec i am sending screenshot !!

**Comment by DhairyaBahl** [comment]
This is my synfig directory ![IMG_20201013_101853.jpg](https://user-images.githubusercontent.com/71627983/95816792-8a562680-0d3d-11eb-8c55-5331442c91ee.jpg)![IMG_20201013_101941.jpg](https://user-images.githubusercontent.com/71627983/95816823-a063e700-0d3d-11eb-85f6-9115b117e97a.jpg) and other picture is the built synfigstudio.exe

**Comment by Keyikedalube** [comment]
@DhairyaBahl Delete your `cmake-build-msys` folder and build synfig again.

**Comment by Keyikedalube** [comment]
@DhairyaBahl Also delete these folders just in case you want to build with other scripts:
- _debug
- cmake-build-debug

**Comment by DhairyaBahl** [comment]
@Keyikedalube what will happen then ?? Did i built it wrong ?? If i will give the same commands again, it will again give me same thing as result..So should i change the commands or something ??

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okay i will delete these and i will build synfig again..but what is the result that we are expecting ?

**Comment by Keyikedalube** [comment]
@DhairyaBahl Did you use this command to build earlier? 2-build-cmake.sh

**Comment by DhairyaBahl** [comment]
Yes !! @Keyikedalube

**Comment by Keyikedalube** [comment]
OK.

Just delete your `cmake-build-msys` and `cmake-build-debug`
And use that command `2-build-cmake.sh` to build again

Now I'm sure you'll see all the warning message pop up
Good luck :)

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okay got it. I will do that. :+1: I will do this and ping you. Hope u dint mind :)

Thanks :)

**Comment by DhairyaBahl** [comment]
@Keyikedalube On which shell i have to build it mingw32 mingw64 or msys2, i have 64 bit OS ??

Don't wanna make any mistakes so just to be sure. Don't mind 

Thanks :)

**Comment by ice0** [comment]
@DhairyaBahl it didn't show warnings because compiler doesn't rebuild files if they not changed.
So the only option in that situation is to delete compiled object file and compile it again. That's why you need to remove build folder.
In your situation (Windows/CMake) better to make a script with this content (inside synfig folder):

**test-cmake-build.sh**
```
#!/usr/bin/env bash

export PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:/opt/mlt-6.16.0/lib/pkgconfig"
mkdir -p cmake-build-msys2 && cd cmake-build-msys2

cmake -G"MSYS Makefiles" ..
make -j4
```

and run it. It should create `cmake-build-msys2` folder, and run build process.

Using `MSYS Makefiles` generator is preferably because it did colored output for warnings and errors, so this will be much easier for you to see them.

**Comment by DhairyaBahl** [comment]
@Keyikedalube I followed ur method !! I got pop-up msgs of all the warnings on console like warning: tool_cutout_icon.sif:<real>:7898: Layer 'warp' rejected value for parameter 'horizon' and others also but there are no wreorder warnings !!

So, how to find those warnings ? 

Thanks :)

**Comment by DhairyaBahl** [comment]
@ice0 Should i still create it, I've completed the build process ??

Thanks for help :)

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube If you guys want i can also send all the pop-up msgs that came on the screen shell !! 

Thanks :)

**Comment by ice0** [comment]
> Should i still create it, I've completed the build process ??

As you wish. But it will be be much cleaner output, when from `2-build-cmake.sh`

For example, this is output I got from it:

```
[ 54%] Building CXX object synfig-studio/src/synfigapp/CMakeFiles/synfigapp.dir/actions/valuedescbake.cpp.obj
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.cpp:35:
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.h: In constructor 'synfigapp::Action::ValueDescCreateChildBone::ValueDescCreateChildBone()':
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.h:56:33: warning: 'synfigapp::Action::ValueDescCreateChildBone::scalelx' will be initialized after [-Wreorder]
   56 |  synfig::ValueBase origin,angle,scalelx,width,tipwidth;
      |                                 ^~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.h:56:27: warning:   'synfig::ValueBase synfigapp::Action::ValueDescCreateChildBone::angle' [-Wreorder]
   56 |  synfig::ValueBase origin,angle,scalelx,width,tipwidth;
      |                           ^~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.cpp:68:1: warning:   when initialized here [-Wreorder]
   68 | Action::ValueDescCreateChildBone::ValueDescCreateChildBone():
      | ^~~~~~
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.cpp:35:
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.h:57:7: warning: 'synfigapp::Action::ValueDescCreateChildBone::c_parent' will be initialized after [-Wreorder]
   57 |  bool c_parent;
      |       ^~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.h:56:41: warning:   'synfig::ValueBase synfigapp::Action::ValueDescCreateChildBone::width' [-Wreorder]
   56 |  synfig::ValueBase origin,angle,scalelx,width,tipwidth;
      |                                         ^~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/synfigapp/actions/valuedesccreatechildbone.cpp:68:1: warning:   when initialized here [-Wreorder]
   68 | Action::ValueDescCreateChildBone::ValueDescCreateChildBone():
      | ^~~~~~
[ 54%] Building CXX object synfig-studio/src/synfigapp/CMakeFiles/synfigapp.dir/actions/valuedescbonesetparent.cpp.obj
```

```
[ 86%] Building CXX object synfig-studio/src/gui/CMakeFiles/synfigstudio.dir/cellrenderer/cellrenderer_time.cpp.obj
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp:60:
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.h: In constructor 'studio::WorkArea::WorkArea(etl::loose_handle<synfigapp::CanvasInterface>)':
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.h:209:7: warning: 'studio::WorkArea::onion_skin' will be initialized after [-Wreorder]
  209 |  bool onion_skin;
      |       ^~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.h:177:7: warning:   'bool studio::WorkArea::highlight_active_bone' [-Wreorder]
  177 |  bool highlight_active_bone;
      |       ^~~~~~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp:124:1: warning:   when initialized here [-Wreorder]
  124 | WorkArea::WorkArea(etl::loose_handle<synfigapp::CanvasInterface> canvas_interface):
      | ^~~~~~~~
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp:60:
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.h:240:23: warning: 'studio::WorkArea::bonesetup_height' will be initialized after [-Wreorder]
  240 |  int bonesetup_width, bonesetup_height;
      |                       ^~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.h:176:39: warning:   'etl::loose_handle<synfig::ValueNode> studio::WorkArea::active_bone_' [-Wreorder]
  176 |  etl::loose_handle<synfig::ValueNode> active_bone_;
      |                                       ^~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp:124:1: warning:   when initialized here [-Wreorder]
  124 | WorkArea::WorkArea(etl::loose_handle<synfigapp::CanvasInterface> canvas_interface):
      | ^~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp: In member function 'bool studio::WorkArea::on_drawing_area_event(GdkEvent*)':
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp:1472:8: warning: this 'else' clause does not guard... [-Wmisleading-indentation]
 1472 |      } else
      |        ^~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/workarea.cpp:1488:3: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
 1488 |   break;
      |   ^~~~~
[ 86%] Building CXX object synfig-studio/src/gui/CMakeFiles/synfigstudio.dir/cellrenderer/cellrenderer_timetrack.cpp.obj
```

```
[ 89%] Building CXX object synfig-studio/src/gui/CMakeFiles/synfigstudio.dir/docks/dialog_tooloptions.cpp.obj
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.cpp:34:
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h: In constructor 'studio::VectorizerSettings::VectorizerSettings(Gtk::Window&, etl::handle<synfig::Layer_Bitmap>, etl::handle<studio::Instance>, std::unordered_map<std::__cxx11::basic_string<char>, int>&, etl::handle<synfig::Layer>)':
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h:90:30: warning: 'studio::VectorizerSettings::instance' will be initialized after [-Wreorder]
   90 |  const etl::handle<Instance> instance;
      |                              ^~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h:62:32: warning:   'Glib::RefPtr<Gtk::Adjustment> studio::VectorizerSettings::adjustment_threshold' [-Wreorder]
   62 |  Glib::RefPtr<Gtk::Adjustment> adjustment_threshold;
      |                                ^~~~~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.cpp:56:1: warning:   when initialized here [-Wreorder]
   56 | VectorizerSettings::VectorizerSettings(Gtk::Window& parent,etl::handle<synfig::Layer_Bitmap> my_layer_bitmap,
      | ^~~~~~~~~~~~~~~~~~
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.cpp:34:
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h:63:18: warning: 'studio::VectorizerSettings::entry_threshold' will be initialized after [-Wreorder]
   63 |  Gtk::SpinButton entry_threshold;
      |                  ^~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h:58:32: warning:   'Glib::RefPtr<Gtk::Adjustment> studio::VectorizerSettings::adjustment_accuracy' [-Wreorder]
   58 |  Glib::RefPtr<Gtk::Adjustment> adjustment_accuracy;
      |                                ^~~~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.cpp:56:1: warning:   when initialized here [-Wreorder]
   56 | VectorizerSettings::VectorizerSettings(Gtk::Window& parent,etl::handle<synfig::Layer_Bitmap> my_layer_bitmap,
      | ^~~~~~~~~~~~~~~~~~
In file included from C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.cpp:34:
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h:65:18: warning: 'studio::VectorizerSettings::entry_despeckling' will be initialized after [-Wreorder]
   65 |  Gtk::SpinButton entry_despeckling;
      |                  ^~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.h:60:32: warning:   'Glib::RefPtr<Gtk::Adjustment> studio::VectorizerSettings::adjustment_accuracy2' [-Wreorder]
   60 |  Glib::RefPtr<Gtk::Adjustment> adjustment_accuracy2;
      |                                ^~~~~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/dialogs/vectorizersettings.cpp:56:1: warning:   when initialized here [-Wreorder]
   56 | VectorizerSettings::VectorizerSettings(Gtk::Window& parent,etl::handle<synfig::Layer_Bitmap> my_layer_bitmap,
      | ^~~~~~~~~~~~~~~~~~
[ 89%] Building CXX object synfig-studio/src/gui/CMakeFiles/synfigstudio.dir/docks/dock_canvases.cpp.obj
```

```
[ 92%] Building CXX object synfig-studio/src/gui/CMakeFiles/synfigstudio.dir/states/state_eyedrop.cpp.obj
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp: In constructor 'studio::StateBone_Context::StateBone_Context(studio::CanvasView*)':
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:114:23: warning: 'studio::StateBone_Context::settings' will be initialized after [-Wreorder]
  114 |  synfigapp::Settings& settings;
      |                       ^~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:107:6: warning:   'int studio::StateBone_Context::active_bone' [-Wreorder]
  107 |  int active_bone;
      |      ^~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:347:1: warning:   when initialized here [-Wreorder]
  347 | StateBone_Context::StateBone_Context(CanvasView *canvas_view) :
      | ^~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:137:19: warning: 'studio::StateBone_Context::radiobutton_skel_deform' will be initialized after [-Wreorder]
  137 |  Gtk::RadioButton radiobutton_skel_deform;
      |                   ^~~~~~~~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:108:6: warning:   'int studio::StateBone_Context::c_layer' [-Wreorder]
  108 |  int c_layer;
      |      ^~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:347:1: warning:   when initialized here [-Wreorder]
  347 | StateBone_Context::StateBone_Context(CanvasView *canvas_view) :
      | ^~~~~~~~~~~~~~~~~
C:/msys64/home/az/projects/synfig/synfig-studio/src/gui/states/state_bone.cpp:449:66: warning: suggest parentheses around arithmetic in operand of '|' [-Wparentheses]
  449 |   get_work_area()->set_type_mask(get_work_area()->get_type_mask()-Duck::TYPE_TANGENT|Duck::TYPE_WIDTH);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
[ 92%] Building CXX object synfig-studio/src/gui/CMakeFiles/synfigstudio.dir/states/state_fill.cpp.obj
```

As you can see, there are a lot of "reordering" warnings.

**Comment by DhairyaBahl** [comment]
I didn't received any warnings on my screen. I guess , i should do the same with your method.

Thanks :)

**Comment by DhairyaBahl** [comment]
,![IMG_20201013_134148.jpg](https://user-images.githubusercontent.com/71627983/95833924-e7f86c00-0d59-11eb-8010-eb981bf49c84.jpg) It is giving some kind of error!!  What is it?

**Comment by DhairyaBahl** [comment]
![IMG_20201013_134413.jpg](https://user-images.githubusercontent.com/71627983/95834150-39a0f680-0d5a-11eb-9906-c937403464e5.jpg)

@ice0 @Keyikedalube

**Comment by ice0** [comment]
@DhairyaBahl you need to update your MSYS2 packages. Where was a bug in `ld` linker which is already fixed.

**Comment by DhairyaBahl** [comment]
@ice0  I will work on this, update the package and then i will run the script again. 

Thanks :+1:

**Comment by DhairyaBahl** [comment]
![IMG_20201013_145735.jpg](https://user-images.githubusercontent.com/71627983/95842687-783bae80-0d64-11eb-8745-1c1e15807b3a.jpg). It says up-to-date

**Comment by ice0** [comment]
Please try
`pacman -Syu`

also show me output of:
`pacman -Ss binutils`

**Comment by DhairyaBahl** [comment]
It's installing few packages,  then I will send u screenshot of pacman binutills command... :) Thanks @ice0

**Comment by DhairyaBahl** [comment]
![IMG_20201013_155625.jpg](https://user-images.githubusercontent.com/71627983/95849161-b63cd080-0d6c-11eb-96fa-5a2a48e0596d.jpg)This is the output of pacman -Ss binutills 

@ice0  Thanks

**Comment by Keyikedalube** [comment]
> @Keyikedalube I followed ur method !! I got pop-up msgs of all the warnings on console like warning: tool_cutout_icon.sif::7898: Layer 'warp' rejected value for parameter 'horizon' and others also but there are no wreorder warnings !!
> 
> So, how to find those warnings ?
> 
> Thanks :)

@DhairyaBahl Strange.

I think your console cleared some output due to the display buffer size limit. I'm not on Windows so I might be very wrong on this guess.

**Comment by Keyikedalube** [comment]
@DhairyaBahl Just wait for ice0 to back get in touch with you about binutils

Otherwise, if I suggest another method you might end up in a deadlock LOL

**Comment by ice0** [comment]
@DhairyaBahl just want to be sure you have latest binutils. Now just remove `cmake-build-msys2` folder, and try this script again. Compilation should be fine.

**Comment by DhairyaBahl** [comment]
@Keyikedalube what is deadlock? 😅

**Comment by DhairyaBahl** [comment]
I did the same @ice0 ..my PC is in recovery mode... scanning and repairing drive C

**Comment by Keyikedalube** [comment]
@DhairyaBahl stuck :) because a computer resource is already occupied

In your case, ice0 is already helping you... and if I jump in before he finishes with you... then you're going to have a hard time deciding between us. You are the resource. And currently, you are occupied :)

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okayy.. got it 🙂

**Comment by DhairyaBahl** [comment]
@ice0 process broke at 78%![IMG_20201013_190826.jpg](https://user-images.githubusercontent.com/71627983/95868078-7e8f5200-0d87-11eb-83ff-05448fe5fd58.jpg)

**Comment by ice0** [comment]
You need to enter `cmake-build-msys2` folder, and run `make -j1` to see where it is broke.

**Comment by DhairyaBahl** [comment]
@ice0 Okay I am doing that then i will send you the output :)

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube Does this always give this much error to every user or i am the unlucky one ?? sweat_smile

**Comment by ice0** [comment]
@DhairyaBahl can you show me output of `git status` command?

**Comment by ice0** [comment]
It looks like your user name contain spaces?
If yes, then I need a time to check that. So I suggest to continue tomorrow.

**Comment by DhairyaBahl** [comment]
@ice0 Okay No probs :) BTW i wanted to ask if i git stash those and then try again, then will it work ??

Thanks :)

**Comment by Keyikedalube** [comment]
> Does this always give this much error to every user or i am the unlucky one ??

🤣😁

**Comment by Keyikedalube** [comment]
> Yes, it contains space!!

Welcome to UNIX world! Space tends to ruin every Windows users life LOL

**Comment by DhairyaBahl** [comment]
@Keyikedalube Shall I Change the user's name if its possible ??

**Comment by Keyikedalube** [comment]
@DhairyaBahl No

**Comment by Keyikedalube** [comment]
@DhairyaBahl Where is your msys installed?

**Comment by DhairyaBahl** [comment]
@Keyikedalube Yes that's where msys2 cloned my project

c:/msys2/home/Dhairya\ Bahl/synfig.git/

**Comment by DhairyaBahl** [comment]
@Keyikedalube Its Dhairya <space> Bahl That's where the problem arise. If i change it, will it result in deadlock ??

**Comment by Keyikedalube** [comment]
@DhairyaBahl Nah

You've got to get used to *nix style. No spaces.

Msys2 is emulate UNIX environment inside your Windows so you've got to get familiar with what works and what doesn't.
Space character is definitely one of the "what doesn't"  in *nix

**Comment by Keyikedalube** [comment]
@DhairyaBahl If you don't want to create another new folder with your name without space "" then I'd suggest moving synfig.git folder to c:/msys2/home OK

It's up to you. Just don't put synfig inside another folder that contain spaces.

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okay so main thing is spaces. got it. btw u and ice0 are how old in this organisation. I mean u guys are pro, when will i reach that level ? :o

**Comment by Keyikedalube** [comment]
@DhairyaBahl I'm just like you... a contributor 😁. I started hacking synfig last year.
So, I'm like 11 months old in this repository.
Me? A pro? Haha. I still got a long way to go...

ice0's been here longer than me.

He's been my guide too when I was stuck in problem solving like you are now.
So he's definitely a pro.

But what matters most is whether you're having **fun** hacking and learning new stuff in this Hacktoberfest event.
Because FUN's where motivation comes from.

I'm noticing you are still new to build tools and Unix environment. Not bad. Open Source software is a good platform to get exposure to real world practical problem solving skills :) It doesn't matter if this issue is too easy. If you don't know the tools then you're in big trouble. Because without the tools you can't get to coding. And you're making a lot of progress. One step at a time. BTW, that's how you get close to becoming a pro.

**Comment by DhairyaBahl** [comment]
@Keyikedalube TBH I was not here for the hacktoberfest event, I thought I am not capable for that,  but later you told me that this repository is hacktoberfest verified, otherwise I was going to leave that 😅😅. 

Yes ice0 is great but you are great 2 ..i am learning from both of you,  although I spent 2 days in just building synfig 😂 but it's better than learning nothing. 

Thanks :)

**Comment by ice0** [comment]
Just created a user with spaces in the name and checked the build.  Nothing. All builds fine (except what I need to install `mingw-cmake` manually), lol )
After what I created another user with non-latin symbols in the name (and with spaces too). Found `Glib::ConvertError`, and I hope to fix it soon.
But that doesn't explain your problem. After that I remember this comment "my PC is in recovery mode... scanning and repairing drive C" and now I think maybe `tool_brush_icon.sif` file just get corrupted?

So can you clone synfig again and try a clean build? (maybe you have a limited internet connection, then I'll try to find another solution)

**Comment by DhairyaBahl** [comment]
@ice0 No.. internet is not a problem for me.. I Will start all over again... and then send you the output of test-build file that u gave me !!

**Comment by DhairyaBahl** [comment]
@ice0 on ur advice I was trying to delete the synfig folder but it wasn't deleting so I moved it to desktop to find out which part of it was not deleting..and this was the file. I thought u should know

![IMG_20201014_115513.jpg](https://user-images.githubusercontent.com/71627983/95951533-366a4100-0e14-11eb-86b6-da8521c07c85.jpg)

**Comment by DhairyaBahl** [comment]
@ice0 Showing this on. /1-setup-windows-msys2.sh![IMG_20201014_122001.jpg](https://user-images.githubusercontent.com/71627983/95953663-c362c980-0e17-11eb-9169-ec8f37f07c21.jpg)

**Comment by ice0** [comment]
@DhairyaBahl just answer no, because this patch is already applied to system header.

**Comment by DhairyaBahl** [comment]
@ice0 yes I did that.. now it's building using test-cmake-msys.sh that u gave

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube successfully built synfig..this time in C:/ without using space in name , now will be working on the issue of wreorder warnings.
 
Thanks :)

**Comment by ice0** [comment]
Resolved by #1750
